controllers/collections.js

```diff
@@ -57,23 +45,49 @@ exports.get = async function (req) {
     }
   }
 
-  const data = await req.db[req.params.collection].find(query, req.query.skip || req.query.offset,
-    req.query.limit, orderby, fields)
-  total = data.length
-  const limit = req.query.pageitems || (total > 10 ? 10 : total)
-  // calculate pagination in case `page`-queryarg was passed
-  const allowed = await Promise.all(data.map((doc) =>
-    util.notify('get', req, req.params.collection, doc).catch(() => false)
-  ))
-  let result = data.filter((doc, i) => allowed[i] === true)
-  if (req.query.page) {
+  if (req.query.page != null) {
+    req.query.page = parseInt(req.query.page)
     if (req.query.page <= 0) {
       throw new util.ApiError(400, 'invalid page number')
     }
-    result = util.createPagination(result, req.query.page, limit)
+    req.query.pageitems = parseInt(req.query.pageitems) || parseInt(req.query.limit) || 10
+    req.query.offset = util.getDatabaseOffset(req.query.page, req.query.pageitems)
+    req.query.orderby = req.query.orderby || util.normalizeOrderBy({ 'meta.created': 1 }) // paging requires orderby to ensure consistent results with offset and limit
+    delete req.query.limit
+    delete req.query.skip
+    let pageData, totalItems
+    if (req.query.pagemetadisable) {
+      // a way to skip the extra count database call, but lose additional page detail
+      pageData = await req.db[req.params.collection].find(query, req.query.offset, req.query.pageitems, req.query.orderby, fields)
+      data = util.createPagePagination(pageData, req.query.page, req.query.pageitems)
+    }
+    else {
+      // this is default - full detail
+      [pageData, totalItems] = await Promise.all([
+        req.db[req.params.collection].find(query, req.query.offset, req.query.pageitems, req.query.orderby, fields),
+        req.db[req.params.collection].count(query)
```

Comment:
I notice we're not passing `skip` or `limit`. I think that makes sense, but maybe we shouldn't even have those parameters on the count methods?

Reply:
i thought long and hard about this too, but since `skip` and `limit` are accepted parameters for `mongo` and `postgres` `count` i figured there must be a scenario outside this where people will use them! we also support them for `find` so might be expected